airflow/providers/amazon/aws/operators/appflow.py

```diff
@@ -0,0 +1,562 @@
+from airflow.operators.python import ShortCircuitOperator
+from airflow.providers.amazon.aws.hooks.appflow import AppflowHook
+
+if TYPE_CHECKING:
+    from mypy_boto3_appflow.client import AppflowClient
+    from mypy_boto3_appflow.type_defs import (
+        DescribeFlowExecutionRecordsResponseTypeDef,
+        ExecutionRecordTypeDef,
+        TaskTypeDef,
+    )
+
+    from airflow.utils.context import Context
+
+EVENTUAL_CONSISTENCY_OFFSET: int = 15  # seconds
+EVENTUAL_CONSISTENCY_POLLING: int = 10  # seconds
+SUPPORTED_SOURCES = {"salesforce", "zendesk"}
+
+
+class AppflowOperatorException(Exception):
+    """Alias for Exception."""
+
+
+class AppflowOperatorBase(BaseOperator):
+    """Amazon Appflow Base Operator class (not supposed to be used directly in DAGs)."""
+
+    BLUE = "#2bccbd"
+    ui_color = BLUE
+
+    def __init__(
+        self,
+        source: str,
+        name: str,
+        flow_update: bool,
+        source_field: Optional[str] = None,
+        dt: Optional[str] = None,
+        poll_interval: int = 20,
+        aws_conn_id: Optional[str] = "aws_default",
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        if source not in SUPPORTED_SOURCES:
+            raise AppflowOperatorException(
+                f"{source} is not a supported source (options: {SUPPORTED_SOURCES})!"
+            )
+        self.dt = dt
+        self._name = name
+        self._source = source
+        self._source_field = source_field
+        self._poll_interval = poll_interval
+        self._aws_conn_id = aws_conn_id
+        self._region = region
+        self._flow_update = flow_update
+
+    @cached_property
+    def hook(self) -> AppflowHook:
+        """Create and return an AppflowHook."""
+        return AppflowHook(aws_conn_id=self.aws_conn_id, region_name=self._region)
+
+    @staticmethod
+    def _dt_to_epoch_str(dt: datetime) -> str:
+        text = str(int(dt.timestamp() * 1000))
+        return text
+
+    def _get_connector_type(self) -> str:
+        connector_type = self._response["sourceFlowConfig"]["connectorType"]
+        if (self.source == "salesforce" and connector_type != "Salesforce") or (
+            self.source == "zendesk" and connector_type != "Zendesk"
+        ):
+            raise AppflowOperatorException(
+                f"Incompatible source ({self.source} and connector type ({connector_type})!"
+            )
+        return connector_type
+
+    def execute(self, context: "Context") -> None:
+        self._af_client: "AppflowClient" = self.hook.conn
+        self._dt_parsed: Optional[datetime] = datetime.fromisoformat(self.dt) if self.dt else None
+        if self._flow_update:
+            self._update_flow()
+        self._run_flow(context)
+
+    def _update_flow(self) -> None:
+        self._response = self._af_client.describe_flow(flowName=self.name)
+        self._connector_type = self._get_connector_type()
```

Comment:
Related to above, that would make these two lines:

```
client = self.hook.conn
response = client.describe_flow(flowName=self.name)
connector_type = self._get_connector_type()
```

which is much easier to read.